src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -557,17 +558,25 @@ private void copyKeyAndTitle() {
                 return;
             }
 
-            Globals.clipboardManager.setContent(sb.toString());
+            final String copiedKeysAndTitles = sb.toString();
+            Globals.clipboardManager.setContent(copiedKeysAndTitles);
 
             if (copied == bes.size()) {
                 // All entries had keys.
-                output((bes.size() > 1 ? Localization.lang("Copied keys") : Localization.lang("Copied key")) + '.');
+                output(Localization.lang("Copied") + " '" + shortenDialogMessage(copiedKeysAndTitles) + "'.");
             } else {
                 output(Localization.lang("Warning: %0 out of %1 entries have undefined BibTeX key.", Integer.toString(bes.size() - copied), Integer.toString(bes.size())));
             }
         }
     }
 
+    private String shortenDialogMessage(String dialogMessage) {
```

Comment:
There is a JavaFX util method that shortens text based on its display size (e.g taking the current font into account). The difference is probably negligible through.

https://github.com/javafxports/openjdk-jfx/blob/develop/modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/skin/Utils.java#L212

Reply:
well, the alternative would be to directly hack a hard limit into the snackbar dialog system. I didn't do that, because maybe another type of messages should be independent of this 300 char limit.